### PR TITLE
Add type hints to `Form` namespace

### PR DESCRIPTION
**Pull request by distantnative**

### CODE REVIEW

File: src/Form/Field/LayoutField.php [hunk 1]
```diff
@@ -219,7 +229,7 @@ protected function setLayouts(array $layouts = []): void
 		);
 	}
 
-	protected function setSettings($settings = null): void
+	protected function setSettings(array|string $settings = null): void
```
Comment (afbora) [inline src/Form/Field/LayoutField.php]:
```suggestion
	protected function setSettings(array|string|null $settings = null): void
```

File: src/Form/FieldClass.php [hunk 2]
```diff
@@ -354,10 +264,8 @@ public function label(): string
 
 	/**
 	 * Returns the parent model
-	 *
-	 * @return mixed
 	 */
-	public function model()
+	public function model(): mixed
```
Comment (afbora) [inline src/Form/FieldClass.php]:
```suggestion
	public function model(): ModelWithContent
```

File: src/Form/FieldClass.php [hunk 3]
```diff
@@ -568,10 +460,8 @@ protected function setTranslate(bool $translate = true): void
 
 	/**
 	 * Setter for the when condition
-	 *
-	 * @param mixed $when
 	 */
-	protected function setWhen($when = null): void
+	protected function setWhen(mixed $when = null): void
```
Comment (afbora) [inline src/Form/FieldClass.php]:
```suggestion
	protected function setWhen(array|null $when = null): void
```

File: src/Form/Mixin/EmptyState.php [hunk 4]
```diff
@@ -4,9 +4,9 @@
 
 trait EmptyState
 {
-	protected $empty;
+	protected string|array|null $empty;
```
Comment (afbora) [inline src/Form/Mixin/EmptyState.php]:
```suggestion
	protected string|null $empty;
```

File: src/Form/Mixin/EmptyState.php [hunk 5]
```diff
@@ -4,9 +4,9 @@
...
 
-	protected function setEmpty($empty = null)
+	protected function setEmpty(string|array|null $empty = null)
```
Comment (afbora) [inline src/Form/Mixin/EmptyState.php]:
```suggestion
	protected function setEmpty(string|array|null $empty = null): string|null
```

**Comment by bastianallgeier** [comment]
@afbora all your changes seem useful to me. @distantnative what do you think? Is it easier to add them through the Github interface or through a manual commit?

**Review by afbora — APPROVED** [review]

**Comment by distantnative** [comment]
Fixed one change that wasn't right. Could be merged now.

**Pull request merged**